File: temp/fix-all-brackets.js
const fs = require('fs');
const path = require('path');

const slidesDir = 'D:/新课开发/党业融合/经营者讲党课/完整课程包/003-授课PPT/slides';
const files = fs.readdirSync(slidesDir).filter(f => f.match(/^slide-\d+\.js$/)).sort();

let fixed = 0;

files.forEach(file => {
  const filePath = path.join(slidesDir, file);
  let content = fs.readFileSync(filePath, 'utf8');
  const original = content;

  // Replace all 「 (U+300C LEFT CORNER BRACKET) with " (U+0022 QUOTATION MARK)
  // Replace all 」 (U+300D RIGHT CORNER BRACKET) with " (U+0022 QUOTATION MARK)
  content = content.replace(/「/g, '"');
  content = content.replace(/」/g, '"');

  if (content !== original) {
    fs.writeFileSync(filePath, content);
    fixed++;
  }
});

console.log(`Fixed ${fixed} files`);

File: temp/fix-all-brackets.py
import re
import os

slidesDir = 'D:/新课开发/党业融合/经营者讲党课/完整课程包/003-授课PPT/slides'
files = sorted([f for f in os.listdir(slidesDir) if f.startswith('slide-') and f.endswith('.js')], key=lambda x: int(re.search(r'\d+', x).group()))

fixed = 0
for fname in files:
    fpath = os.path.join(slidesDir, fname)
    with open(fpath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Replace Chinese brackets with single quotes (they were used as quotation marks in Chinese)
    # But inside JS strings we need to escape them - so replace with \'
    # Actually simpler: replace 「 with ' and 」 with ' 
    new_content = content.replace('「', "'").replace('」', "'")
    
    if new_content != content:
        with open(fpath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        fixed += 1

print(f'Fixed {fixed} files')
